tests/unit_tests/production_configuration/test_generate_production_grid.py

```diff
@@ -0,0 +1,306 @@
+        assert isinstance(point["radius"], Quantity)
+        assert isinstance(point["viewcone"], Quantity)
+
+
+def test_clean_grid_output(grid_gen):
+    # Case 1: Valid grid points
+    grid_points = [
+        {
+            "zenith_angle": 30 * u.deg,
+            "azimuth": 310 * u.deg,
+            "nsb": 4 * u.MHz,
+            "energy_threshold": {"lower": 0.1 * u.TeV},
+            "radius": 100 * u.m,
+            "viewcone": 1 * u.deg,
+        },
+        {
+            "zenith_angle": 40 * u.deg,
+            "azimuth": 345 * u.deg,
+            "nsb": 5 * u.MHz,
+            "energy_threshold": {"lower": 0.2 * u.TeV},
+            "radius": 200 * u.m,
+            "viewcone": 2 * u.deg,
+        },
+    ]
+    cleaned_points = grid_gen.clean_grid_output(grid_points)
+    assert isinstance(cleaned_points, str)  # JSON string
+    assert '"zenith_angle"' in cleaned_points
+    assert '"energy_threshold"' in cleaned_points
+
+
+def test_serialize_quantity(grid_gen, caplog):
+    # Case 1: Value is a Quantity (single value)
+    quantity = 5 * u.m
+    serialized = grid_gen.serialize_quantity(quantity)
+    assert serialized == {"value": 5, "unit": "m"}
+
+    # Case 2: Value is not a Quantity (single value)
+    value = 5
+
+    with caplog.at_level(logging.WARNING):
+        grid_gen.serialize_quantity(value)
+
+    assert "Unsupported type" in caplog.text
+    assert str(type(value)) in caplog.text
+
+
+def test_convert_altaz_to_radec_and_coordinates(grid_gen):
+    # Case 1: Valid AltAz to RA/Dec conversion
+    alt, az = 45.0 * u.deg, 30.0 * u.deg
+    radec = grid_gen.convert_altaz_to_radec(alt, az)
+    assert isinstance(radec, SkyCoord)
+    assert radec.ra.unit == u.deg
+    assert radec.dec.unit == u.deg
+
+    # Case 2: Valid coordinate conversion
+    grid_gen.coordinate_system = "ra_dec"
+    grid_points = [
+        {"zenith_angle": 30 * u.deg, "azimuth": 45 * u.deg},
+        {"zenith_angle": 20 * u.deg, "azimuth": 60 * u.deg},
+    ]
+    converted_points = grid_gen.convert_coordinates(grid_points)
+    assert "ra" in converted_points[0]
+    assert "dec" in converted_points[0]
+
+    # Case 3: Missing zenith_angle or azimuth
+    grid_points = [{"azimuth": 45 * u.deg}]
+    converted_points = grid_gen.convert_coordinates(grid_points)
+    assert "ra" not in converted_points[0]
+    assert "dec" not in converted_points[0]
+
+
+def test_create_circular_binning(grid_gen):
+    # Case 1: No wraparound
+    bins = grid_gen.create_circular_binning((0, 150), 6)
+    assert len(bins) == 6
+    assert bins[0] == 0
+    assert bins[-1] == 150
+    assert bins[1] == 30
+    assert bins[2] == 60
+
+    # Case 2: Wraparound
+    bins = grid_gen.create_circular_binning((300, 20), 5)
+    assert len(bins) == 5
+    assert bins[0] == 300
+    assert bins[-1] == 20
+    assert bins[1] == 320
+
+    # Case 3: Single bin
+    bins = grid_gen.create_circular_binning((0, 360), 1)
+    assert len(bins) == 1
+    assert bins[0] == 0
```

Comment:
Not sure - shouldn't that be the bin center? Would that be 180 for a [0,360] deg bin?

Reply:
OK - misread it - all good.